execution.bs

```diff
@@ -4973,15 +5088,79 @@ enum class forward_progress_guarantee {
+
+        5. The member <code><i>default-impls</i>::<i>get-env</i></code> is initialized
+            with a callable object equal to the following lambda:
+
+              <pre highlight="c++">
+              []&lt;class Rcvr>(auto index, auto& state, const Rcvr& rcvr) noexcept
+                -> decltype(<i>FWD-ENV</i>(execution::get_env(rcvr))) {
+                return <i>FWD-ENV</i>(execution::get_env(rcvr));
+              }
+              </pre>
+
+        6. The member <code><i>default-impls</i>::<i>get-state</i></code> is initialized
+            with a callable object equal to the following lambda:
+
+              <pre highlight="c++">
+              []&lt;class Sndr>(Sndr&& sndr, auto& rcvr) noexcept -> decltype(auto) {
+                return get&lt;1>(std::forward&lt;Sndr>(sndr));
+              }
+              </pre>
+
+        7. The member <code><i>default-impls</i>::<i>start</i></code> is initialized
+            with a callable object equal to the following lambda:
+
+              <pre highlight="c++">
+              [](auto& state, auto& rcvr, auto&... ops) noexcept -> completion_signatures<> {
+                (execution::start(ops), ...);
+                return {};
+              }
+              </pre>
+
+        8. The member <code><i>default-impls</i>::<i>complete</i></code> is initialized
+            with a callable object equal to the following lambda:
+
+              <pre highlight="c++">
+              []&lt;class Index, class Rcvr, class Tag, class... Args>(
+                Index, auto& state, Rcvr& rcvr, Tag, Args&&... args) noexcept
+                  -> completion_signatures&lt;Tag(Args...)>
+                    requires <i>callable</i>&lt;Tag, Rcvr, Args...> {
+                <i>// Mandates: Index::value == 0</i>
+                tag(std::move(rcvr), std::forward&lt;Args>(args)...); 
+                return {};
+              }
+              </pre>
+
+        9. The return type of <code><i>basic-sender</i></code>'s customization of
+            `get_completion_signatures` is computed as follows.
 
-            - The unspecified base type has no non-static data members. It may
-                define member functions or hidden friend functions ([hidden.friends]).
+              1. Let `Env` be the type of an environment, let the type `Sndr` be
+                  the (possibly cv- and ref-qualified) type of a specialization
+                  of <code><i>basic-sender</i></code>, and let
+                  <code>Child<sub><i>i</i></sub></code> be the type of the
+                  <code><i>i</i></code>-th child sender of `Sndr`, with the same
+                  cv- and ref-qualifications as `Sndr`.
 
-            - <span class="wg21note">An expression of type <code><i>basic-sender</i></code> is usable as the
-                initializer of a structured binding declaration
-                [dcl.struct.bind].</span>
+              2. Let <code>ChildEnv<sub><i>i</i></sub></code> be the type 
```

Comment:
This para is incomplete. I haven't settled on suitable wording yet. It wants to appeal to the blanket wording about completion signatures in [exec.snd.general].

Something like:

> _Remarks:_ For a subexpression `sndr` let `Sndr` be `decltype((sndr))`. Let `rcvr` be a receiver whose associated environment has type `Env`. Returns a specialization of `completion_signatures` the set of whose template arguments correspond to the set of completion operations that are partially evaluated as a result of calling `start` on the operation state that results from connecting `sndr` and `rcvr`.

???